frontend/src/pages/editProfile.js

```diff
@@ -1,40 +1,131 @@
-import React from "react";
-import { useSelector, useDispatch } from "react-redux";
+import React, { useEffect, useState } from "react";
 import { useNavigate } from "react-router-dom";
 import Button from "react-bootstrap/Button";
 import Form from "react-bootstrap/Form";
 import Container from "react-bootstrap/Container";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
 
-import { editProfile } from "../store/profile";
 import InputField from "../components/input";
-import { validate, validateField, getErrorPayload } from "../utils/validation";
+import { validate, validateField } from "../utils/validation";
+import * as profileService from "../services/profileService";
+import { getErrorMessage } from "../utils/validation";
 
 const EditProfile = () => {
-  const state = useSelector((state) => state.editProfile.value);
-  const dispatch = useDispatch();
+  const [profile_id, setProfile_id] = useState();
+  const [user_id, setUser_id] = useState();
+  const [success, setSuccess] = useState(false);
+  const [first_name, setFirst_name] = useState("");
+  const [last_name, setLast_name] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [password2, setPassword2] = useState("");
+  const [errors, setErrors] = useState({});
+  const state = {
+    profile_id: profile_id,
+    user_id: user_id,
+    success: success,
+    first_name: first_name,
+    last_name: last_name,
+    email: email,
+    password: password,
+    password2: password2,
+    errors: errors,
+  };
+
   const navigate = useNavigate();
+  let avatarFile = {};
+
+  useEffect(() => {
+    const fetchProfile = async () => {
+      const { data } = await profileService.getCurrentProfile();
+      setProfile_id(data.profile.id);
+      setUser_id(data.profile.user_id);
+      setFirst_name(data.user.first_name);
+      setLast_name(data.user.last_name);
+      setEmail(data.user.email);
+    };
+    fetchProfile();
```

Comment:
just remove the function, it is extra step unless you are using it more than once
```suggestion  
    //place at the top
     const [profile, setProfile] = useState();
     
     const { data } = await profileService.getCurrentProfile()
     
     //then you can just do this one liner, and have a object state
      setProfile(data);
      
      //this is how you call
      { profile.first_name }
```